.github/workflows/release.yml

```diff
@@ -0,0 +1,28 @@
+name: Release
+on:
+  workflow_dispatch:
+    branches:
+      - main
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # This makes action fetch all Git history so that Changesets can generate changelogs with the correct commits
+          fetch-depth: 0
+      - name: Use Node.js 14.x
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+      - name: Install Dependencies
+        run: npm install
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }} # Ensure to have this set up under GitHub secrets
+      - name: Create Release Pull Request or Publish to npm
+        uses: changesets/action@v1
+        with:
+          publish: npm run release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Ensure to have this set up under GitHub secrets
```

Comment:
You don't set `GITHUB_TOKEN` in secrets. Instead, you add `permissions` object on this workflow job, and then `GITHUB_TOKEN` will be automatically available with those permissions. Docs: https://docs.github.com/en/actions/security-guides/automatic-token-authentication

Reply:
I was referring to the comment, which can be removed. And also I didn't see a `permissions` section in the yaml at the time, but maybe the default permissions are fine.